packages/iov-keycontrol/src/keyring-entries/slip10.ts

```diff
@@ -227,8 +226,20 @@ export class Slip10KeyringEntry implements KeyringEntry {
     prehashType: PrehashType,
     _: ChainId,
   ): Promise<SignatureBytes> {
-    const keypair = await this.privkeyForIdentity(identity);
-    const signature = await Ed25519.createSignature(prehash(transactionBytes, prehashType), keypair);
+    const privkey = await this.privkeyForIdentity(identity);
+    const message = prehash(transactionBytes, prehashType);
+
+    let signature: Uint8Array;
+    switch (this.curve) {
+      case Slip10Curve.Ed25519:
```

Comment:
I think whether we want to pull out this switch into protected method from subclass, but honestly overkill if we only add a new curve once every blue moon....

Reply:
Only one more curve [is defined in SLIP-0010](https://github.com/satoshilabs/slips/blob/master/slip-0010.md#master-key-generation): NIST P-256 curve. If we want to have it, no problem. But I don't think there is more coming in the foreseeable future